### PR TITLE
PR: Improve content and organization of history pane in docs

### PR DESCRIPTION
**Pull request by juanis2112**
### Pull Request Checklist

* [ ] Read and followed this repo's [Contributing Guidelines](https://github.com/spyder-ide/spyder/blob/master/CONTRIBUTING.md)
* [ ] Based your PR on the latest version of the correct branch (master or 3.x)
* [ ] Checked your writing carefully for correct English spelling, grammar, etc
* [ ] Described your changes and the motivation for them below
* [ ] Noted what issue(s) this pull request resolves, creating one if needed

### Issue(s) Resolved

Fixes #

### CODE REVIEW

File: doc/historylog.rst
Comment (CAM-Gerlach) [inline doc/historylog.rst:10]:
```suggestion
```

* ~Grammar: "for" -> "with"~
* Unless its a placeholder for an image that isn't there at all, I would suggest that you use the OP, Github comments or elsewhere for meta remarks like this, so they don't pollute the repo itself and cause unnecessary problems when rebasing. Thanks.

File: doc/historylog.rst
Comment (CAM-Gerlach) [inline doc/historylog.rst:7]: Per @juanis2112 :

> Replace screenshot for screenshot without the options menu

File: doc/historylog.rst
Comment (CAM-Gerlach) [inline doc/historylog.rst:19]:
```suggestion
Each Spyder session is marked by a date and timestamp, making it easy to remember when you executed a certain command.
```

My mistake

File: doc/historylog.rst
Comment (CAM-Gerlach) [inline doc/historylog.rst:21]:
```suggestion
Just like in the Editor, selecting a word or phrase displays all other occurrences, and full syntax highlighting is also supported.
```

Avoid repetition in my original text

File: doc/historylog.rst
Comment (CAM-Gerlach) [inline doc/historylog.rst:16]: Needs "pane" here to be clear that its referring to the pane, rather than the history shown by the pane, a usage which matches e.g. the Find pane docs.

```suggestion
======================
Using the History pane
======================
```

File: doc/historylog.rst
Comment (CAM-Gerlach) [inline doc/historylog.rst:28]:
```suggestion

The top-right options menu (:guilabel:`Hamburger` icon) allows you to set the number of commands that the :guilabel:`History` pane should remember.
```

Needs an update for Spyder 4, a line break and a small grammar tweak

File: doc/historylog.rst
Comment (CAM-Gerlach) [inline doc/historylog.rst:32]:
```suggestion
You can also toggle wrapping of long lines (:guilabel:`Wrap lines`), and whether the line number is displayed to the left of the text (:guilabel:`Show line numbers`).
```

Simplify syntax, avoid ambiguity and be consistent about stating the guilabel name,

File: doc/historylog.rst
Comment (CAM-Gerlach) [inline doc/historylog.rst:42]:
```suggestion
The list of commands shown in the :guilabel:`History` pane are stored in :file:`history.py` in the :file:`.spyder-py3` directory in your user home folder (by default, :file:`C:/Users/{username}` on Windows, :file:`/Users/{username}` for macOS, and typically :file:`/home/{username}` on GNU/Linux).
```

Add a few details, and be more clear

File: doc/historylog.rst
Comment (CAM-Gerlach) [inline doc/historylog.rst:45]: Is it really necessary to show this? Its just the same file that's shown in the history log.

**Review by CAM-Gerlach — CHANGES_REQUESTED** [review]
Some textual suggestions and a comment on one of the proposed GIFs

As a sidenote, can we decide on a consistent convention for denoting image/GIF placeholders (I would suggest brackets, `[]`, as they are generally what are used), and stick to it?

**Review by ccordoba12 — COMMENTED** [review]
Thanks @juanis2112! I left a small suggestion for you, otherwise looks good.

File: doc/historylog.rst
Comment (CAM-Gerlach) [reply doc/historylog.rst:46]: I guess...but what's the practical use case? Would anyone likely actually use this?

File: doc/historylog.rst
Comment (CAM-Gerlach) [reply doc/historylog.rst:43]: Using `/` for paths consistently across platforms is what is specified in the style guide and implemented across the current docs for both [paths in general](https://github.com/spyder-ide/spyder-docs/blob/master/STYLEGUIDE.md#writing-all-languages) and the [the file role specifically](https://github.com/spyder-ide/spyder-docs/blob/master/STYLEGUIDE.md#rest-style-elements). Given most paths mentioned in the docs are not platform-specific, it makes sense to just pick one path separator and use it consistently, rather than treating specially the small number of absolute paths that are unique to Windows.

Furthermore, in Python source code `/` is used almost universally for paths regardless of platform (by convention with string paths, and by requirement with `pathlib` `Path`s); indeed, our own copy relative/absolute path commands copy them this way, so it will avoid things breaking things if users copy them into their code, the console. Finally, given this usage, users are users across platforms are sufficiently used to `/` in paths in Python code, URLs and elsewhere that the likelihood of confusion by using `/` on Windows is effectively zero; despite many places in Spyder's UI where `/` is or was used on Windows, the only cases where this has genuinely caused an issue for even beginners was when `/` and `\` were mixed in the _same_ path or otherwise used inconsistently, which this would help avoid.

Comment (ccordoba12) [reply doc/historylog.rst:43]: > Using / for paths consistently across platforms is what is specified in the style guide

Ok, I wasn't aware of that. Then we can leave it as it is.

File: doc/historylog.rst
Comment (ccordoba12) [inline doc/historylog.rst:29]: I think this gif should be easier to understand by activating first `Show line numbers` and then `Wrap lines`.

File: doc/historylog.rst
Comment (ccordoba12) [inline doc/historylog.rst:40]: Could you use the light macOS theme? The dark theme is non-standard and a bit hard to read.

**Review by ccordoba12 — COMMENTED** [review]
Thanks @juanis2112 for the new screenshots and gifs! I left two comments for you about them.

File: doc/historylog.rst
Comment (CAM-Gerlach) [inline doc/historylog.rst]: A line break was mistakenly removed here (3 before second-level headers).

File: doc/historylog.rst
Comment (CAM-Gerlach) [inline doc/historylog.rst]: A spurious extra line break was added here.

File: doc/historylog.rst
Comment (CAM-Gerlach) [inline doc/historylog.rst:7]: This screenshot looks good, but `optipng` reduces the size of this 0.5 MB image by nearly 50%, entirely for free (aside from a few seconds of runtime). I can do it while rebasing before the merge, if desired.

File: doc/historylog.rst
Comment (CAM-Gerlach) [reply doc/historylog.rst:29]: Agreed. Otherwise, LGTM.

File: doc/historylog.rst
Comment (CAM-Gerlach) [reply doc/historylog.rst:40]: - [ ] Its an OS, not Spyder level setting on a minority OS, and likely a majority of developers prefer the dark theme, so I don't see the "non-standard" nature of the dark theme (built-in to the OS and heavily promoted by Apple) as an issue. However, I do agree the text for invisible items is quite hard to read with it enabled, so I guess its worth changing for this particular screenshot.
- [ ] It would be much better to show the file in a real editor with syntax highlighting, so its clearer its just a normal Python file rather than some bespoke text format.
- [ ] The size of the screenshot, at nearly 2 MB, is gargantuan for what it shows, larger than most GIFs. Downscaling by 2x and applying `optipng` results in a size of only 0.35 MB, less than 1/5 of the original, for essentially no loss in perceptible quality at the final displayed resolution. Again, to make things simplest, I can do this as part of the rebase pass once you've updated the screenshot accordingly.

**Review by CAM-Gerlach — CHANGES_REQUESTED** [review]
My comments on the screenshot/GIF changes.

File: doc/historylog.rst
Comment (ccordoba12) [reply doc/historylog.rst:40]: > Its an OS, not Spyder level setting on a minority OS, and likely a majority of developers prefer the dark theme, so I don't see the "non-standard" nature of the dark theme

I disagree. It took me a long time (i.e. several seconds) to understand what that screenshot was about. I didn't get that a Finder instance was displayed at the left until I paid close attention to the window the details.

Also, please remember that our docs are for people that are just getting started with Spyder. Of course it can be useful for experienced developers too, but that's not our main target.

Finally, macOS can have a small market share but most tutorials and videos are done with it. So I think most people are accustomed to see Mac screenshots here and there.

> However, I do agree the text for invisible items is quite hard to read with it enabled, so I guess its worth changing for this particular screenshot.

Yep, that's another important issue.

> It would be much better to show the file in a real editor with syntax highlighting

Agreed. You can use Sublime for that @juanis2112.

> The size of the screenshot, at nearly 2 MB, is gargantuan for what it shows

Good point too.

Comment (CAM-Gerlach) [reply doc/historylog.rst:40]: > I disagree. It took me a long time (i.e. several seconds) to understand what that screenshot was about. I didn't get that a Finder instance was displayed at the left until I paid close attention to the window the details.

Okay, then if it is an issue with familiarity, that makes sense then. Thanks for explaining.

File: doc/historylog.rst
Comment (CAM-Gerlach) [reply doc/historylog.rst:29]: This was addressed

File: doc/historylog.rst
Comment (CAM-Gerlach) [reply doc/historylog.rst:7]: Will be addressed on rebase

File: doc/historylog.rst
Comment (CAM-Gerlach) [reply doc/historylog.rst:40]: Its using a real editor now, but the session timestamp (`############ Date ##########` that the history log adds) is not visible anymore, so its not clear its a history log file. I would suggest showing that. Also, it looks much clearer to me what's going on, though as far as I can tell, it looks like its still using the dark theme so I'm not sure if @ccordoba12 still thinks it should be changed (but I would assume yes).

**Review by CAM-Gerlach — COMMENTED** [review]
Just one thing, besides what we agreed I'll fix in the rebase. @ccordoba12 please take note not to merge these PRs until I've rebased them, thanks.

File: doc/historylog.rst
Comment (ccordoba12) [reply doc/historylog.rst:40]: > I would suggest showing that

Agreed.

> I'm not sure if @ccordoba12 still thinks it should be changed (but I would assume yes).

Correct, I still prefer to see Finder on the left with the Mac light theme.

Comment (CAM-Gerlach) [reply doc/historylog.rst:40]: @juanis2112 Should be last two things on this PR, other than the line break and image stuff I'll fix:

* [ ] Scroll editor to the top of the file to show the session timestamp, etc.
* [ ] Use light OS theme

File: doc/historylog.rst
Comment (CAM-Gerlach) [inline doc/historylog.rst:43]: Reminder to remove this spurious line break on rebase

File: doc/historylog.rst
Comment (CAM-Gerlach) [inline doc/historylog.rst:33]: Remember to fix on rebase

File: doc/historylog.rst
Comment (CAM-Gerlach) [reply doc/historylog.rst:40]: LGTM now.

**Review by CAM-Gerlach — APPROVED** [review]
LGTM now, just waiting on final approval and rebase.

**Review by ccordoba12 — APPROVED** [review]
Thanks @juanis2112! Great job!

**Comment by CAM-Gerlach** [comment]
@ccordoba12 Rebase and merge in work.

**Pull request merged**
